### PR TITLE
fix allocate command that changed for 5.x

### PR DESCRIPTION
**Pull request by jcantrill**
This command replaces the one in 2.x since the syntax changed and means something else.

Note: Additional changes should allow this script to be pulled from the image and used in conjunction with an oc binary on a host node

### CODE REVIEW

File: elasticsearch/utils/allocate-stale-primary [hunk 1]
```diff
@@ -15,36 +15,34 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 set -euo pipefail
-source es_util_env
 
-shard=0
-allow_primary=true
+if [ -z "${OC_CMD:-''}" ] ; then
```
Comment (ewolinetz) [inline elasticsearch/utils/allocate-stale-primary]: how would we use `OC_CMD`? What would it be set to?

Comment (richm) [reply elasticsearch/utils/allocate-stale-primary]: you don't need the `''` - you can just use `[ -z "${OC_CMD:-}" ]`

Comment (richm) [reply elasticsearch/utils/allocate-stale-primary]: For example, if you had a copy of the script outside of the pod, on the master host, you could run
```
OC_CMD="oc exec -c elasticsearch $espod" ./allocate_stale_primary index ...
```

Comment (jcantrill) [reply elasticsearch/utils/allocate-stale-primary]: I have found that some of these utils are useful outside the context of the pod.  This allows you to either use on pod or extract and use on host like:

``
DC_NAME=logging-es-data-master-f26an8nv OC_CMD="oc exec -c elasticsearch $pod --" SHARD=0 ./allocate-stale-primary .searchguard
```
This allows use without needing to modify it

File: elasticsearch/utils/move-replica-shard [hunk 2]
```diff
@@ -15,9 +15,10 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 set -euo pipefail
-source es_util_env
 
-allow_primary=true
+if [ -z "${OC_CMD:-''}" ] ; then
```
Comment (ewolinetz) [inline elasticsearch/utils/move-replica-shard]: same here `''` not needed

File: elasticsearch/README.md [hunk 3]
```diff
@@ -14,15 +14,26 @@ Example to retrieve the list of nodes
...
+a primary shard.
 
 Example:
 ```
-$ oc exec -c elasticsearch $POD -- allocate-shard .kibana.02c55f18a892b365bcd1802db9e5c9df39c04674
+$ oc exec -c elasticsearch $POD -- allocate-replica .kibana.02c55f18a892b365bcd1802db9e5c9df39c04674
+```
+
+### allocate-stale-primary
+Manually allocate-stale-primary shard to a given node in the Elasticsearch cluster.  The node
```
Comment (richm) [inline elasticsearch/README.md]: Manually allocate a stale primary shard?

File: elasticsearch/README.md [hunk 4]
```diff
@@ -14,15 +14,26 @@ Example to retrieve the list of nodes
...
 
-### allocate-shard
-Manually allocate a shard to a given node in the Elasticsearch cluster.  The node
+### allocate-replica
+Manually a replica shard to a given node in the Elasticsearch cluster.  The node
 by default will be the one on which the command is executed. This command
-is used to explicitly [reroute and allocate](https://www.elastic.co/guide/en/elasticsearch/reference/2.4/cluster-reroute.html)
-a shard
+is used to explicitly [reroute and allocate](https://www.elastic.co/guide/en/elasticsearch/reference/5.6/cluster-reroute.html#cluster-reroute)
+a primary shard.
```
Comment (richm) [inline elasticsearch/README.md]: a replica shard?

**Comment by ewolinetz** [comment]
@jcantrill looks like theres a few docs nits on this

**Comment by jcantrill** [comment]
@ewolinetz updated

**Pull request merged**